activerecord/lib/active_record/model_schema.rb

```diff
@@ -439,6 +439,29 @@ def reset_column_information
         initialize_find_by_cache
       end
 
+      # Sets the implicit order column records are sorted by when an ordered finder
+      # method such as +first+ or +last+ is called but without using an explicit
+      # order clause.
+      #
+      # Example:
+      #
+      #   class Project < ActiveRecord::Base
+      #     self.implicit_order_column = "created_at"
+      #   end
+      #
+      # By default the implicit order column is the primary key.
+      #
+      # Note that using a non-unique column may result in non-deterministic results.
+      def implicit_order_column=(column_name)
+        @implicit_order_column = column_name.to_s
+      end
+
+      # Returns the name of the column records will be ordered by if no explicit
+      # order clause is used. Defaults to the primary key.
+      def implicit_order_column
+        @implicit_order_column ||= primary_key
+      end
```

Comment:
Instead of doing this as a getter / setter, could we just let subclasses override?

For example, the implementation in active record is:

```ruby
class ActiveRecord::Base
  def self.implicit_order_column
    primary_key
  end
end
```

If you want to override it in your subclass, just do:

```ruby
class Project < ActiveRecord::Base
  def self.implicit_column_order
    "created_at"
  end
end
```

That way we don't have to deal with ivar / cache management.

Reply:
I think `class_attribute` would be better: it should be as simple as the override version, but is much more accessible for e.g. a future deprecation.

Reply:
I did look into `class_attribute` but it wasn't immediately obvious how I'd set the default value to `primary_key`.

Reply:
@eileencodes / @tenderlove any thoughts on this? should I update to use `class_attribute` instead?

Reply:
@tekin yes, make this a class attribute and then we'll merge it. 😊

Reply:
Updated to use `class_attribute`.